File: server/Models/Book.cs
using System.Collections.Generic;

namespace CollectorsVault.Server.Models
{
    public class Book : VaultItem
    {
        /// <summary>List of author names.</summary>
        public List<string> Authors { get; set; } = new List<string>();
        public string? ISBN { get; set; }
        public string? Publisher { get; set; }
        /// <summary>Publish date text as returned/provided (e.g. "September 21, 1937", "1997").</summary>
        public string? PublishDateString { get; set; }
        public int? PageCount { get; set; }
        public string? CoverSmall { get; set; }
        public string? CoverMedium { get; set; }
        public string? CoverLarge { get; set; }
        public string? BookUrl { get; set; }
        /// <summary>Subject / genre tags from the lookup provider.</summary>
        public List<string>? Subjects { get; set; }
        /// <summary>Physical or digital format of the book.</summary>
        public BookFormat? BookFormat { get; set; }
        /// <summary>Indicates whether this copy needs to be replaced.</summary>
        public bool NeedsReplacement { get; set; }
        /// <summary>Name of the series this book belongs to, if any.</summary>
        public string? SeriesName { get; set; }
        /// <summary>Position of this book within its series, if known.</summary>
        public int? SeriesNumber { get; set; }
    }
}

File: client/src/models/Book.ts
export type BookFormat =
    | 'Unknown'
    | 'Hardcover'
    | 'Paperback'
    | 'MassMarketPaperback'
    | 'TradePaperback'
    | 'BoardBook'
    | 'LibraryBinding'
    | 'SpiralBound'
    | 'EBook'
    | 'Audiobook'
    | 'Other';

export interface BookLookupResult {
    title: string;
    authors: string[];
    isbn: string;
    publisher: string;
    publishDate: string;
    pageCount?: number;
    description: string;
    subjects: { key: string; value: string }[];
    coverSmall: string;
    coverMedium: string;
    coverLarge: string;
    providerUrl: string;
    seriesName: string;
    seriesNumber?: number;
    seriesNotFound: boolean;
    bookFormat?: BookFormat;
}

export interface Book {
    title: string;
    authors: string[];
    isbn?: string;
    publisher?: string;
    publishDateString?: string;
    pageCount?: number;
    description?: string;
    subjects?: string[];
    coverSmall?: string;
    coverMedium?: string;
    coverLarge?: string;
    bookUrl?: string;
    bookFormat?: BookFormat;
    needsReplacement?: boolean;
    seriesName?: string;
    seriesNumber?: number;
}
